action.yml

```diff
@@ -83,7 +83,27 @@ runs:
             process.exit(1);
           }
 
+    - name: Set Condition Variables
+      id: condition-vars
+      uses: actions/github-script@v7
+      env:
+        PAYLOAD_ARG: ${{ inputs.client_payload }}
+      with:
+        script: |
+          try {
+            const payload = JSON.parse(process.env.PAYLOAD_ARG);
+            const isNonCommitEvent = payload.hasOwnProperty('isNonCommitEvent') ? payload.isNonCommitEvent : 'false';
```

Comment:
why not `const isNonCommitEvent = payload.isNonCommitEvent === true`?
whats the benefit of `hasOwnProperty` in this case?